### ISSUE TITLE
Segmentation fault at parser/mpParserBase.cpp:1158

### ISSUE THREAD

**Issue by nasailja** [beltoforion/muparserx#58]
The expression `{?{{{{:44` leads to:
```
Program received signal EXC_BAD_ACCESS, Could not access memory.
Reason: KERN_INVALID_ADDRESS at address: 0x0000000000000000
mup::ParserXBase::ParseFromRPN (this=0x7fff5fbff7c0) at parser/mpParserBase.cpp:1158
1158	        if (pStack[sidx--]->GetBool()==false)
(gdb) up
#1  0x0000000100206834 in ?? ()
(gdb) 
#2  0x00000001000013f0 in main () at muparserx.cpp:17
17			double result = parser.Eval().GetFloat();
```
Same parser as in https://github.com/beltoforion/muparserx/issues/55

**Comment by beltoforion** [comment]
Nice tool. I'll check your reports. Suprising to see that its still finding so many things i was under the assumption that the the syntax checking is pretty solid.

**Comment by nasailja** [comment]
> Nice tool. I'll check your reports. Suprising to see that its still
> finding so many things i was under the assumption that the the syntax
> checking is pretty solid.

Heh yeah afl has a long list of bugs found in gnupg, openssh, etc. and 
in less critical software. The expressions did get quite weird but it 
would be nice if muparserx didn't assert or segfault.

**Comment by beltoforion** [comment]
I agree. I take assertions and segfaults personally in my software

**Comment by beltoforion** [comment]
Fixed in Version 4.0.1
